src/jdbc/cubrid/jdbc/jci/UStatement.java

```diff
@@ -2317,16 +2317,20 @@ private void readATupleByOid(CUBRIDOID oid, UInputBuffer inBuffer)
 
 	private void readATuple(int index, UInputBuffer inBuffer)
 	        throws UJciException {
-		tuples[index] = new UResultTuple(inBuffer.readInt(), columnNumber);
-		tuples[index].setOid(inBuffer.readOID(relatedConnection.getCUBRIDConnection()));
+
+		UResultTuple tuple = new UResultTuple(inBuffer.readInt(), columnNumber);
+		int n = tuple.tupleNumber() - 1;
+		tuples[n] = tuple;
```

Comment:
this line can throw the OutOfIndexException. the `tuples` is allocated as the size of `fetchSize`, and the `tupleNumber()` can be over the `fetchSize`.

Reply:
tuples is allocated with TotalTupleNumber which is the number of whole result tuples not of a fetched tuples.

Reply:
It seems to slip in #2443.
```
// read_fetch_data ()
		if (tuples == null) {
			if (totalTupleNumber > 0) {
				tuples = new UResultTuple[totalTupleNumber];
			}
		}
```
It may cause OutOfMemoryError if the number of whole result's row is big enough.

Reply:
for result cache the whole results should be cached, however tuples is allocated first empty array not any data assigned.
in allocation step, OutOfMemoryError may not occur.

Reply:
read_fetch_data () is not only for result cache, it is used in fetching resultset normally. and even data is not assigned, OutOfMemoryError could occur in the allocation step too.

And this change seems to break the main concept: JDBC only receives and processes `fetch_size` results in a single request.
-Allocates an array to hold the result (UResultTuple) as much as `fetch_size`.
-Reads (parsing) the contents of the requested buffer and makes it UResultTuple.
-After reading all `fetch_size`, JDBC will re-request and reallocate the result array.

If `the tuples` is not reallocated, UResultTuple will not be deleted by GC and then OutOfMemoryError will occur eventually.

Reply:
When the  cursor is closed or statement is closed, the tuples array is free.

If we allocate the tuples as fetch size only, is there any way how to keep the result cache? and the performance of result-cache can be degraded. If we wouldn't find a way to keep the cache, we may not use the result cache as policy.